File: Coding Challenges/DailyChallenge/JavaScript/20260129_LettersNumbers.js
function separateLettersAndNumbers(str) {
  const regEx = /((?<=[a-z])[0-9]|(?<=[0-9])[a-z])/i

  let newString = '';
  let prevChar = '';

  for(let i = 0; i < str.length; i++) {
    let char = str[i];

    if((prevChar + char).match(regEx)) {
      newString += `-${char}`;
    }
    else {
      newString += char;
    }

    prevChar = char;
  }

  return newString;
}

File: Coding Challenges/Daily Challenge/2026/January/20260129-Letters-Numbers.py
import re

def separate_letters_and_numbers(s):
    reg_ex = '((?<=[a-z])[0-9]|(?<=[0-9])[a-z])'

    new_string = ''
    prev_char = ''

    for char in s:
        if re.search(reg_ex, prev_char + char, re.IGNORECASE) :
            new_string += f'-{char}'
        else:
            new_string += char
        
        prev_char = char

    return new_string